platform-sdk/swirlds-virtualmap/src/jmh/java/com/swirlds/virtualmap/benchmark/reconnect/VirtualMapReconnectBenchBase.java

```diff
@@ -0,0 +1,323 @@
+import com.swirlds.common.merkle.MerkleInternal;
+import com.swirlds.common.merkle.synchronization.config.ReconnectConfig;
+import com.swirlds.common.merkle.synchronization.config.ReconnectConfig_;
+import com.swirlds.common.merkle.synchronization.internal.QueryResponse;
+import com.swirlds.common.test.fixtures.merkle.dummy.DummyMerkleInternal;
+import com.swirlds.common.test.fixtures.merkle.dummy.DummyMerkleLeaf;
+import com.swirlds.common.test.fixtures.merkle.util.MerkleTestUtils;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.extensions.test.fixtures.TestConfigBuilder;
+import com.swirlds.metrics.api.Metrics;
+import com.swirlds.virtualmap.VirtualMap;
+import com.swirlds.virtualmap.datasource.VirtualDataSource;
+import com.swirlds.virtualmap.datasource.VirtualDataSourceBuilder;
+import com.swirlds.virtualmap.datasource.VirtualHashRecord;
+import com.swirlds.virtualmap.datasource.VirtualLeafRecord;
+import com.swirlds.virtualmap.internal.merkle.VirtualMapState;
+import com.swirlds.virtualmap.internal.merkle.VirtualRootNode;
+import com.swirlds.virtualmap.internal.pipeline.VirtualRoot;
+import com.swirlds.virtualmap.test.fixtures.InMemoryBuilder;
+import com.swirlds.virtualmap.test.fixtures.TestKey;
+import com.swirlds.virtualmap.test.fixtures.TestValue;
+import org.junit.jupiter.api.Assertions;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.swirlds.common.test.fixtures.io.ResourceLoader.loadLog4jContext;
+
+/**
+ * The code is largely borrowed from VirtualMapReconnectTestBase.java in swirlds-virtualmap/src/test/.
+ * Ideally, it belongs to a shared test fixture, but I was unable to find a way to resolve dependencies
+ * between projects and modules, so I created this copy here and removed a few static definitions that
+ * are irrelevant to JMH benchmarks. In the future, this JMH-specific copy may in fact diverge
+ * from the unit test base class if/when we implement performance testing-related features here
+ * (e.g. artificial latencies etc.)
+ */
+public abstract class VirtualMapReconnectBenchBase {
+
+    protected VirtualMap<TestKey, TestValue> teacherMap;
+    protected VirtualMap<TestKey, TestValue> learnerMap;
+    protected BrokenBuilder teacherBuilder;
+    protected BrokenBuilder learnerBuilder;
+
+    protected final Configuration configuration = new TestConfigBuilder().getOrCreateConfig();
+    protected final ReconnectConfig reconnectConfig = configuration.getConfigData(ReconnectConfig.class);
+
+    protected VirtualDataSourceBuilder<TestKey, TestValue> createBuilder() {
+        return new InMemoryBuilder();
+    }
+
+    protected void setupEach() {
+        final VirtualDataSourceBuilder<TestKey, TestValue> dataSourceBuilder = createBuilder();
+        teacherBuilder = new BrokenBuilder(dataSourceBuilder);
```

Comment:
Why do we need BrokenBuilder here? This is a performance benchmark, not a unit test to check how resilient virtual map reconnects are to I/O or network issues

Reply:
I removed the unnecessary entities.